client/src/components/BlockLinkField/BlockLinkField.js

```diff
@@ -58,7 +65,91 @@ jQuery.entwine('ss', ($) => {
 
         $(`#${internalDialogId}`).updateModalTitle($(this).find(`#${linkName}_Title`).val());
       }
-    }
+    },
+
+    /**
+     * Registering a change will replace the BlockLinkField content with a holding message
+     * to encourage the user to save the form to proceed with the new link data.
+     */
+    registerChange() {
+      this.find('.blocklinkfield__content')
+        .empty()
+        .append(
+          $('<span/>')
+            .addClass('blocklinkfield__content--message')
+            .text(i18n._t('BlockLinkField.ModifiedMessage', 'Changes will be visible upon save'))
+        );
+    },
+  });
+
+  $('.js-injector-boot .blocklinkfield__actions').entwine({
+    /**
+     * Get the BlockLinkField holder that owns the current actions popover
+     *
+     * @return {Object}
+     */
+    getLinkField() {
+      return this.parent('.blocklinkfield');
+    },
+
+    /**
+     * Get the hidden JSON data field from the containing BlockLinkField
+     *
+     * @return {Object}
+     */
+    getLinkDataField() {
+      return this.getLinkField().find('input:hidden.blocklinkfield');
+    },
+
+    /**
+     * If the BlockLinkField has some data set, render the actions popover to allow users
+     * to clear it.
+     */
+    onmatch() {
+      const data = JSON.parse(this.getLinkDataField().val());
+
+      if (data && typeof data.PageID !== 'undefined') {
+        this.renderActionsMenu();
+      }
+    },
+
+    /**
+     * Gather the actions for the popover menu and render it
+     */
+    renderActionsMenu() {
+      const actions = CONSTANTS.LINK_ACTIONS.map((action) => {
+        if (!action.callback) {
+          switch (action.value) {
+            case 'clear': {
+              return {
+                ...action,
+                callback: () => {
+                  // Remove the hidden JSON data object
+                  this.getLinkDataField().val('{}');
+
+                  // Register a change on the field, switching the content to a holding message
+                  this.getLinkField().registerChange();
+
+                  // Close the popover (NOTE DOESN'T WORK YET)
+                  this.remove();
+                }
+              };
+            }
+            default: {
+              return action;
+            }
+          }
+        }
+        return action;
+      });
+
+      ReactDOM.render(
+        <InjectedBlockLinkFieldActions
+          actions={actions}
```

Comment:
I think it's a good idea to pass a `container` property here and the value could be `$('.cms-content')[0]` or `this.closest('form')[0]` then you can pass this through to the Popover component.

This will help alleviate overflow issues with the bottom-bar or preview panel to the right (if there is going to be one in the future).